internal/impl/postgresql/pglogicalstream/replication_message_decoders.go

```diff
@@ -148,7 +150,28 @@ func decodePgOutput(WALData []byte, relations map[uint32]*RelationMessage, typeM
 
 func decodeTextColumnData(mi *pgtype.Map, data []byte, dataType uint32) (interface{}, error) {
 	if dt, ok := mi.TypeForOID(dataType); ok {
-		return dt.Codec.DecodeValue(mi, dataType, pgtype.TextFormatCode, data)
+		val, err := dt.Codec.DecodeValue(mi, dataType, pgtype.TextFormatCode, data)
+		if err != nil {
+			return val, err
+		}
+
+		if dt.Name == "uuid" {
+			typesValueForUUID := val.([16]uint8)
```

Comment:
This will panic if the cast fails. Can we also check if the type is correct and return an error if not?